src/model.ts

```diff
@@ -0,0 +1,82 @@
+import {Random} from "@reside-ic/random";
+
+import {BaseType} from "./base";
+
+/**
+ * Constructor for a {@link DustModel}
+ *
+ * @param base The singleton base object
+ *
+ * @param pars Parameters to pass to the model
+ *
+ */
+export type DustModelConstructable = new(base: BaseType, pars: any) => DustModel;
+
+/**
+ * Information returned by an initialised model about itself
+ */
+export type DustModelInfo = DustModelVariable[];
+
+/**
+ * Information about a single variable within a model, typically found
+ * within {@link DustModelInfo}
+ */
+export interface DustModelVariable {
+    /**
+     * Dimensions of the variable.
+     * * empty vector - a scalar
+     * * length-1 vector - length of a vector
+     * * length-2 vector - number of rows and columns of a matrix
+     * * length-n vector - size of each dimension
+     *
+     * Note that as the product of an empty vector is 1,
+     * so the product of `dim` is always equal to `length`
+     */
+    dim: number[];
+
+    /**
+     * Length of the variable - if the variable represents a vector or
+     * multidimensional array, then this is the length over all
+     * dimensions. Scalars have length 1, so that the sum of all
+     * length elements within an {@link DustModelInfo} array
```

Comment:
this last sentence doesn't quite scan, missing a word?

Reply:
so often I write docs, then that makes me realise I've made a design mistake, so I stop mid-sentence and fix the mistake but forget to come back and sort it out. Emma found one of these too I think